fog/view/server/src/sharding_strategy.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Enables a Fog View Store to know for which blocks to process TxOuts.
+//!
+//! By determining which TxOuts to process, we are able to "shard" the set of
+//! TxOuts across Fog View Store instances.
+
+use mc_blockchain_types::BlockIndex;
+use mc_fog_types::common::BlockRange;
+
+/// Tells a Fog View Store for which blocks it should process TxOuts.
+pub trait ShardingStrategy {
+    /// Returns true if the Fog View Store should process this block.
+    fn should_process_block(&self, block_index: BlockIndex) -> bool;
+}
+
+/// Determines whether or not to process a block's TxOuts based on the "epoch"
+/// sharding strategy, in which a block is processed IFF it falls within the
+/// contiguous range of blocks.
+///
+/// In practice, the set of Fog View Shards will contain overlapping
+/// [epoch_block_ranges] in order to obfuscate which shard processed the TxOuts.
+pub struct EpochShardingStrategy {
+    /// If a block falls within this range, then the Fog View Store should
+    /// process its TxOuts.
+    epoch_block_range: BlockRange,
+}
+
+impl ShardingStrategy for EpochShardingStrategy {
+    fn should_process_block(&self, block_index: BlockIndex) -> bool {
+        self.epoch_block_range.contains(block_index)
+    }
+}
+
+impl Default for EpochShardingStrategy {
+    fn default() -> Self {
+        Self {
+            epoch_block_range: BlockRange::new(0, u64::MAX),
+        }
+    }
+}
+
+impl EpochShardingStrategy {
+    #[allow(dead_code)]
+    pub fn new(epoch_block_range: BlockRange) -> Self {
+        Self { epoch_block_range }
+    }
+}
+
+#[cfg(test)]
+mod epoch_sharding_strategy_tests {
+    use super::*;
+
+    #[test]
+    fn should_process_block_block_index_is_before_epoch_start_returns_false() {
+        const START_BLOCK: BlockIndex = 50;
+        const END_BLOCK_NON_INCLUSIVE: BlockIndex = 100;
```

Comment:
nit: I don't think it matters too much, curious if `END_BLOCK_EXCLUSIVE` was intentionally avoided as a name.

Reply:
Done.